src/psyclone/tests/psyad/adjoint_visitor_test.py

```diff
@@ -413,6 +443,151 @@ def test_assignment_node(fortran_reader, fortran_writer):
+    assert isinstance(ad_loop, Loop)
+
+    with pytest.raises(VisitorError) as info:
+        _ = adj_visitor.loop_node(tl_loop)
+    assert ("A passive loop node should not be processed by the loop_node() "
+            "method within the AdjointVisitor() class, as it should have been "
+            "dealt with by the schedule_node() method." in str(info.value))
+
+
+@pytest.mark.parametrize("in_bounds,out_bounds", [
+    ("lo,hi", "hi, lo, -1"),
+    ("lo,hi,1", "hi, lo, -1"),
+    ("lo,hi,-1", "hi - MOD(hi - lo, - 1), lo, 1"),
+    ("lo,hi,step", "hi - MOD(hi - lo, step), lo, -1 * step")])
+def test_loop_node_active(fortran_reader, fortran_writer, in_bounds,
+                          out_bounds):
+    '''Test that a loop_node containing active variables returns with its
+    loop order reversed and its loop body processed by the adjoint
+    visitor. Checks that appropriate offset code is generated when the
+    loop step is not, or might not be, 1.
+
+    '''
+    code = TL_LOOP_CODE.replace("lo,hi,step", in_bounds)
+    tl_psyir = fortran_reader.psyir_from_source(code)
+    tl_loop = tl_psyir.walk(Loop)[0]
+    assert isinstance(tl_loop, Loop)
+    adj_visitor = AdjointVisitor(["a", "b", "c"])
+    ad_psyir = adj_visitor(tl_psyir)
+    ad_loop = ad_psyir.walk(Loop)[0]
+    assert isinstance(ad_loop, Loop)
+    result = fortran_writer(ad_loop)
+    expected_result = (
+        "do i = {0}\n"
+        "  b(i) = b(i) + a(i)\n"
+        "  c(i) = c(i) + a(i)\n"
+        "  a(i) = 0.0\n"
+        "enddo\n".format(out_bounds))
+    assert result == expected_result
+
+
+@pytest.mark.xfail(reason="issue #1235: caplog returns an empty string in "
+                   "github actions.", strict=False)
+def test_loop_logger(fortran_reader, caplog):
+    '''Test that the logger writes the expected output if the loop_node
+    method is called with an inactive node and an active node.
+
+    '''
+    tl_psyir = fortran_reader.psyir_from_source(TL_LOOP_CODE)
+    tl_loop = tl_psyir.walk(Loop)[0]
+    assert isinstance(tl_loop, Loop)
+
+    adj_visitor = AdjointVisitor(["a", "b", "c"])
+    # Need to use _visit() here rather than adj_visitor(tl_psyir) as
+    # the latter takes a copy of the tree in case any lowering needs
+    # to be done and that causes the stored symbols for active
+    # variables to be different which means they do not match in
+    # subsequent calls. This only a problem for tests as we don't
+    # normally call loop_node() or similar, directly.
+    _ = adj_visitor._visit(tl_psyir)
+
+    # active loop
+    with caplog.at_level(logging.INFO):
+        _ = adj_visitor.loop_node(tl_loop)
+    assert caplog.text == ""
+    with caplog.at_level(logging.DEBUG):
+        _ = adj_visitor.loop_node(tl_loop)
+    assert "Transforming active loop" in caplog.text
+
+    # Remove content for subsequent inactive loop code
+    caplog.clear()
+
+    # inactive loop
+    adj_visitor = AdjointVisitor(["d", "e"])
+    _ = adj_visitor(tl_psyir)
```

Comment:
I must be missing something but again, why is this call required?

Reply:
I added a comment here too.